components/views/card-view.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * A generic card view class.
+ *
+ * Subclasses can override any of the following:
+ * - configuration (string or function): Sets the display style of the card as square or list. Can take values of
+ *      "square_card" or "list_card". Defaults to "square_card".
+ * - action (function): Action to take when the action button is clicked. Defaults to a no-op.
+ * - cardClass (string or function): Class name for this card's DOM element. Defaults to the empty string.
+ * - pennant (string or function): Text of the card's pennant. No pennant is displayed if this value is falsy.
+ * - title (string or function): Title of the card. Defaults to the empty string.
+ * - description (string or function): Description of the card. Defaults to the empty string.
+ * - details (array or function): Array of child views to be rendered as details of this card. The class "meta-detail"
+ *      is automatically added to each rendered child view to ensure appropriate styling. Defaults to an empty list.
+ * - actionClass (string or function): Class name for the action DOM element. Defaults to the empty string.
+ * - actionUrl (string or function): URL to navigate to when the action button is clicked. Defaults to the empty string.
+ * - actionContent: Content of the action button. This may include HTML. Default to the empty string.
+ */
+;(function (define) {
+    'use strict';
+    define(['jquery',
+            'underscore',
+            'backbone',
+            'text!templates/views/card-view/basic-card-view.underscore',
+            'text!templates/views/card-view/square-card-view.underscore'],
+        function ($, _, Backbone, basicCardViewTemplate, squareCardViewTemplate) {
+            var CardView = Backbone.View.extend({
+                tagName: 'li',
+
+                events: {
+                    'click .action' : 'action'
```

Comment:
Currently this isn't doing anything because the cards are links. If we were to update and add an action other than a link then we should use a button instead of an anchor tag. Is there a current use case for cards doing something different on click other than linking to a page?